lib/ar_state_machine.rb

```diff
@@ -83,7 +83,7 @@ def save_state_change
   end
 
   def do_state_change_before_callbacks
-    return false unless self.class.run_before_transition_callbacks(self.state, self, old_state)
+    return false if self.class.run_before_transition_callbacks(self.state, self, old_state) == false
```

Comment:
does it ever return nil?

Reply:
yea, that was actually the issue we were having.